cmake/install/FindThrust.cmake

```diff
@@ -0,0 +1,76 @@
+##============================================================================
+##  Copyright (c) Kitware, Inc.
+##  All rights reserved.
+##  See LICENSE.txt for details.
+##  This software is distributed WITHOUT ANY WARRANTY; without even
+##  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+##  PURPOSE.  See the above copyright notice for more information.
```

Comment:
What is the license for this?

Reply:
Yeah, that's a good point. I guess it's this: https://gitlab.kitware.com/vtk/vtk-m/-/blob/783867eeb05e0a6538f9c520af02c3615651b4ed/LICENSE.txt. It does say 

```
Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are
met:

 * Redistributions of source code must retain the above copyright
   notice, this list of conditions and the following disclaimer.

 * Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the
   distribution.

 * Neither the name of Kitware nor the names of any contributors may
   be used to endorse or promote products derived from this software
   without specific prior written permission.

```
So it seems pretty permissive, but IANAL.

I can write a custom one from scratch if this causes issues. Chances are it would look very similar, though, since really, the only thing that is not boilerplate in there is the version parsing, and that comes from [Nvidia](https://github.com/NVIDIA/thrust/blob/main/thrust/cmake/README.md).

Reply:
I think it's the cmake license, not VTK. This is the BSD 3-clause license.  That is one of the licenses we can easily use. We still have a process we need to follow to add a new piece of software. I will check if RAPIDS already has done this for CMake code.

Reply:
https://gitlab.kitware.com/cmake/cmake (readme says BSD 3-clause)

Reply:
OK, we have approval, but before we can merge we need the complete license for this file included (ideally *in this file*).

Reply:
Ok, so I copied in the full text from the LICENSE.txt above.